include/SDL3/SDL_process.h

```diff
@@ -0,0 +1,180 @@
+#define SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN          "SDL.process.create.stdout"
+#define SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN          "SDL.process.create.stderr"
+#define SDL_PROP_PROCESS_CREATE_REDIRECT_STDERR_BOOLEAN "SDL.process.create.redirect_stderr"
+
+/**
+ * Get the properties associated with a process.
+ *
+ * The following read-only properties are provided by SDL:
+ *
+ * - `SDL_PROP_PROCESS_PID_NUMBER`: the process ID of the process.
+ * - `SDL_PROP_PROCESS_STDIN_POINTER`: an SDL_IOStream that can be used to write input to the process, if it was created with `SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN` set to true.
+ * - `SDL_PROP_PROCESS_STDOUT_POINTER`: an SDL_IOStream that can be used to read output from the process, if it was created with `SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN` set to true.
+ * - `SDL_PROP_PROCESS_STDERR_POINTER`: an SDL_IOStream that can be used to read error output from the process, if it was created with `SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN` set to true.
+ *
+ * \param process the process to query.
+ * \returns a valid property ID on success or 0 on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \sa SDL_CreateProcess
+ * \sa SDL_CreateProcessWithProperties
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC SDL_PropertiesID SDL_GetProcessProperties(SDL_Process *process);
+
+#define SDL_PROP_PROCESS_PID_NUMBER     "SDL.process.pid"
+#define SDL_PROP_PROCESS_STDIN_POINTER  "SDL.process.stdin"
+#define SDL_PROP_PROCESS_STDOUT_POINTER "SDL.process.stdout"
+#define SDL_PROP_PROCESS_STDERR_POINTER "SDL.process.stderr"
+
+/**
+ * Stop a process.
+ *
+ * \param process The process to stop.
+ * \param force Stop the process immediately, without giving it a chance to free its resources.
+ * \returns SDL_TRUE on success or SDL_FALSE on failure; call SDL_GetError()
+ *          for more information.
+ *
+ * \sa SDL_CreateProcess
+ * \sa SDL_CreateProcessWithProperties
+ * \sa SDL_WaitProcess
+ * \sa SDL_DestroyProcess
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_KillProcess(SDL_Process *process, SDL_bool force);
+
+/**
+ * Wait for a process to finish.
+ *
+ * This can be called multiple times to get the status of a process.
+ *
+ * \param process The process to wait for.
+ * \param block If true, block until the process finishes; otherwise, report on the process' status.
+ * \param exitcode a pointer filled in with the process exit code if the process has exited, may be NULL.
+ * \returns SDL_TRUE if the process exited, SDL_FALSE otherwise.
+ *
+ * \sa SDL_CreateProcess
+ * \sa SDL_CreateProcessWithProperties
+ * \sa SDL_KillProcess
+ * \sa SDL_DestroyProcess
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC SDL_bool SDLCALL SDL_WaitProcess(SDL_Process *process, SDL_bool block, int *exitcode);
```

Comment:
I think `exitcode` should become a SDL_process property, that is only available after we have waited on the child process to finish.

Reply:
Done!